drivers/core_demographics_report/app/models/core_demographics_report/details.rb

```diff
@@ -35,6 +35,7 @@ def detail_hash
     def detail_scope_from_key(key)
       detail = detail_hash[key]
       return report_scope.none unless detail
+      return report_scope.none if ! detail[:can_view_details].nil? && ! detail[:can_view_details]
```

Comment:
This _should_ handle both cases a bit more cleanly.  
```suggestion
      return report_scope.none unless detail&.try(:[], :can_view_details)
```

If `detail` isn't set, it will return nil which will evaluate to false, if it's false, it'll be false, if it's true, it'll be true.

Reply:
The issue is the can_view_details is currently only set on the disability calculations so the other calculations' details won't populate because the value is nil. But this is a lot cleaner and maybe the better route is adding this and a can_view_details that defaults to true for all of the other details sections in this report

Reply:
Looking at it more, it may be easier to jsut run this in the controller and redirect back to the report if the user tries to load details they don't have access too.